### ISSUE TITLE
[FEATURE REQUEST] Gemeni benchmarks

### ISSUE THREAD

**Issue by kim-borgen** [ScandEval/ScandEval#443]
### 🚀 The feature, motivation and pitch

I would like to benchmark google's Gemeni models trough their API on scandinavic languages. 
(Or preferably, benchmark any model trough any API that fits a standardized spec, they are all pretty equal)

I can do the actual benchmarking myself, but Google's API needs to be integrated.

Why?
- We are considering using Google's gemeni model for our AI systems.
- Gemini 1.5 Pro and Flash provide magnitudes more context length and lower cost than OpenAI alternatives.
- Googles model are available in Europe for enterprise use before OpenAI-Azure. (Gemeni 1.5 Pro is the best model available for enterprise use in Europe based on the lmsys leaderboard)
- Google has magnitudes more compute available and will thus probably take the lead soon.

**Comment by saattrupdan** [comment]
Makes sense! In that case we probably  need to create a `gemini_models` module, analogous to `openai_models`, which has classes `GeminiTokenizer` and `GeminiModel`, as well as a `model_setups.gemini` module analogous to `model_setups.openai`, which sorts out model existence, model loading and model config.

If there is a large amount of overlap between that and OpenAI then we can think about refactoring, but I don't know the Google API well enough. Maybe you can clarify this, @kim-borgen?

**Comment by kim-borgen** [comment]
I have not used the google API/sdk before, but seems like a piece of cake. Probably easier to use the sdk  https://ai.google.dev/gemini-api

I haven't looked too deep in the scandeval code, but is all the setup, tokenizer, and helper classes really necessary for API models like openai and google? 

Actually, when I think of it, maybe the easiest approach would be to just implement a LlamaIndex wrapper, thus we can easily benchmark a lot of closed models including gemeni, openAI, claude, and mistral. Example for gemeni: https://docs.llamaindex.ai/en/stable/examples/llm/gemini/

**Comment by saattrupdan** [comment]
I'm not familiar with Llamaindex, but evaluating the closed models currently involve more than "merely" generating sequences. JSON mode is heavily used for NER tasks, and we use the logits directly for 4/7 tasks (this is where the tokenizer is used as well).

I'm all for a common API wrapper, but that requires that they all work in the same way, and I kind of doubt that this is the case. OpenAI keeps changing theirs (they had logprobs included previously, then removed them, and now they're back again, and JSON mode is new as well), and the tokenizers are probably different as well (I don't suppose everyone use tiktoken).

So my guess is that such helper classes are necessary, unfortunately.

**Comment by kim-borgen** [comment]
Yeah no, seems like gemeni does not output logprobs at all https://ai.google.dev/api/python/google/generativeai/types/ChatResponse

But its fine to skip tasks that require logits/logprobs i guess 🤷 Benchmark what can be benchmarked :) 

However, LlamaIndex's openai wrapper does output logprobs https://docs.llamaindex.ai/en/stable/api_reference/llms/openai/

Yeah, maybe scratch the API wrapper idea, but a llamaindex wrapper may be a way to go.

Either way, the gemeni sdk does have a json mode 
```
genai.GenerativeModel("gemini-pro").generate_content(
    self.prompt,
    generation_config=GenerationConfig(response_mime_type="application/json")
)
```
Also possible to do it in LlamaIndex by supplying this GenerationConfig on initialization. LlamaIndex also has a nice pydantic `structured_output`, but I don't think Gemeni support this yet, so I created an issue https://github.com/run-llama/llama_index/issues/13840
